xclim/indices/generic.py

```diff
@@ -475,6 +477,83 @@ def spell_length_statistics(
     return to_agg_units(out, data, "count")
 
 
+@declare_relative_units(thresh="<data>")
+def season(
+    data: xarray.DataArray,
+    thresh: Quantified,
+    window: int,
+    op: str,
+    stat: str,
+    freq: str,
+    mid_date: DayOfYearStr | None = None,
+    constrain: Sequence[str] | None = None,
+) -> xarray.DataArray:
+    r"""Season
+
+    A season starts when a variable respects some condition for a consecutive run of `N` days. It stops
+    when the condition is inverted for `N` days. Runs where the condition is not met for fewer than `N` days
+    are thus allowed. Additionally a middle date can serve as a maximal start date and minimum end date.
+
+    Parameters
+    ----------
+    data : xarray.DataArray
+        Variable.
+    thresh : Quantified
+        Threshold on which to base evaluation.
+    window : int
+        Minimum number of days that the condition must be met / not met for the start / end of the season.
+    op : str
+        Comparison operation.
+    stat : {'start', 'end', 'length'}
+        Which season facet to return.
+    freq : str
+        Resampling frequency.
+    mid_date : DayOfYearStr, optional
+        An optional middle date to restrict the possible start and end of the season.
```

Comment:
here or below in a "Notes" section, it would be good to refer to `season_start/season_end`. The notion of `middle_date` is not super clear just by reading the `season` docstring

Reply:
Done!
`mid_date` for all season functions.

Rewrote doc.